kapitan/refs/secrets/vaultkv.py

```diff
@@ -0,0 +1,295 @@
+                raise Exception('Neither VAULT_CACERT nor VAULT_CAPATH specified')
+            client_parameters['verify'] = cert_path
+        else:
+            client_parameters['verify'] = cert
+    else:
+        client_parameters['verify'] = False
+
+    # CLIENT CERTIFICATE FOR TLS AUTHENTICATION
+    client_key = parameter.get('VAULT_CLIENT_KEY', os.getenv('VAULT_CLIENT_KEY',default=''))
+    client_cert = parameter.get('VAULT_CLIENT_CERT', os.getenv('VAULT_CLIENT_CERT',default='')) 
+    if client_key != '' and client_cert != '':
+        client_parameters['cert'] = (client_cert,client_key)
+    return client_parameters
+
+def vault_obj(vault_parameters):
+    """
+    vault_parameters: necessary parameters to authenticate & get value from vault, provided by inventory
+    e.g.:
+        auth: userpass
+        VAULT_ADDR: http://127.0.0.1:8200
+        VAULT_SKIP_VERIFY: false
+    Authenticate client to server and return client object
+    """
+    env = get_env(vault_parameters)
+
+    client = hvac.Client(
+        **{k:v for k,v in env.items() if k is not 'auth'}
+    )
+
+    auth_type = vault_parameters['auth']
+    # GET TOKEN EITHER FROM ENVIRONMENT OF FILE
+    if auth_type in ['token','github']:
+        env['token'] = os.getenv('VAULT_TOKEN')
+        if not env['token']:
+            try:
+                token_file = os.path.join(os.path.expanduser('~'),'.vault-token')
+                with open(token_file,'r') as f:
+                    env['token'] = f.read()
+                if env['token'] == '':
+                    raise VaultError('{file} is empty'.format(file=token_file))
+            except IOError:
+                VaultError("Cannot read file {file}".format(file=token_file))
+    # DIFFERENT LOGIN METHOD BASED ON AUTHENTICATION TYPE
+    if auth_type == 'token':
+        client.token = env['token']
+    elif auth_type == 'ldap':
+        client.auth.ldap.login(
+            username = os.getenv('VAULT_USERNAME'),
+            password = os.getenv('VAULT_PASSWORD')
+        )
+    elif auth_type == 'userpass':
+        client.auth_userpass(
+            username=os.getenv('VAULT_USERNAME'),
+            password=os.getenv('VAULT_PASSWORD')
+        )
+    elif auth_type == 'approle':
+        client.auth_approle(
+            os.getenv('VAULT_ROLE_ID'),
+            secret_id=os.getenv('VAULT_SECRET_ID')
+        )
+    elif auth_type == 'github':
+        client.auth.github.login(token=env['token'])
+    else:
+        raise "Authentication type '{auth}' not supported".format(auth=auth_type)
+
+    if client.is_authenticated():
+        return client
+    else:
+        VaultError("Vault Authentication Error, Environment Variables defined?")
+
+
+class VaultSecret(Base64Ref):
+    """
+    Hashicorp Vault support for KV Secret Engine
+    """
+    def __init__(self, data, **kwargs):
+        """
+        Set vault parameter and encoding of data
+        """
+        self.data = data.rstrip()
```

Comment:
How come you `rstrip()` data here? Can you add a comment above and also explain what the data format is at that point?